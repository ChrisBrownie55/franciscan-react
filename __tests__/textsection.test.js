/* eslint-env jest */
import React from 'react'
import { mount } from 'enzyme'

import TextSection from '../components/TextSection'

describe('TextSection', () => {
  it('should be defined', () => {
    expect(TextSection).toBeDefined()
  })
  it('should render', () => {
    const wrapper = mount(<TextSection />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render text when passed in', () => {
    const wrapper = mount(<TextSection text="Test text" />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render a title when passed in', () => {
    const wrapper = mount(<TextSection sectionTitle="Test Title" />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render a title when passed in', () => {
    const wrapper = mount(<TextSection textColor="#000" />)
    expect(wrapper).toMatchSnapshot()
  })
})
