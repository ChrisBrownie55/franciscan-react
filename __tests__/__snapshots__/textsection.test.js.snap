// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSection should render 1`] = `
<WithStyles(TextSection)>
  <TextSection
    classes={
      Object {
        "heroBtn": "TextSection-heroBtn-4",
        "innerContainer": "TextSection-innerContainer-2",
        "invertedBtn": "TextSection-invertedBtn-5",
        "section": "TextSection-section-1",
        "sectionText": "TextSection-sectionText-3",
        "white": "TextSection-white-6",
      }
    }
  >
    <div
      className="TextSection-section-1"
      style={
        Object {
          "backgroundColor": undefined,
          "borderBottom": "2px solid undefined",
          "borderTop": "2px solid undefined",
        }
      }
    >
      <div
        className="TextSection-innerContainer-2"
      >
        <WithStyles(Typography)
          gutterBottom={true}
          style={
            Object {
              "color": undefined,
            }
          }
          variant="display2"
        >
          <Typography
            align="inherit"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter-20",
                "alignJustify": "MuiTypography-alignJustify-22",
                "alignLeft": "MuiTypography-alignLeft-19",
                "alignRight": "MuiTypography-alignRight-21",
                "body1": "MuiTypography-body1-16",
                "body2": "MuiTypography-body2-15",
                "button": "MuiTypography-button-18",
                "caption": "MuiTypography-caption-17",
                "colorError": "MuiTypography-colorError-30",
                "colorInherit": "MuiTypography-colorInherit-26",
                "colorPrimary": "MuiTypography-colorPrimary-27",
                "colorSecondary": "MuiTypography-colorSecondary-28",
                "colorTextSecondary": "MuiTypography-colorTextSecondary-29",
                "display1": "MuiTypography-display1-11",
                "display2": "MuiTypography-display2-10",
                "display3": "MuiTypography-display3-9",
                "display4": "MuiTypography-display4-8",
                "gutterBottom": "MuiTypography-gutterBottom-24",
                "headline": "MuiTypography-headline-12",
                "noWrap": "MuiTypography-noWrap-23",
                "paragraph": "MuiTypography-paragraph-25",
                "root": "MuiTypography-root-7",
                "subheading": "MuiTypography-subheading-14",
                "title": "MuiTypography-title-13",
              }
            }
            color="default"
            gutterBottom={true}
            headlineMapping={
              Object {
                "body1": "p",
                "body2": "aside",
                "display1": "h1",
                "display2": "h1",
                "display3": "h1",
                "display4": "h1",
                "headline": "h1",
                "subheading": "h3",
                "title": "h2",
              }
            }
            noWrap={false}
            paragraph={false}
            style={
              Object {
                "color": undefined,
              }
            }
            variant="display2"
          >
            <h1
              className="MuiTypography-root-7 MuiTypography-display2-10 MuiTypography-gutterBottom-24"
              style={
                Object {
                  "color": undefined,
                }
              }
            />
          </Typography>
        </WithStyles(Typography)>
        <WithStyles(Typography)
          className="TextSection-sectionText-3"
          gutterBottom={true}
          style={
            Object {
              "color": undefined,
            }
          }
          variant="body1"
        >
          <Typography
            align="inherit"
            className="TextSection-sectionText-3"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter-20",
                "alignJustify": "MuiTypography-alignJustify-22",
                "alignLeft": "MuiTypography-alignLeft-19",
                "alignRight": "MuiTypography-alignRight-21",
                "body1": "MuiTypography-body1-16",
                "body2": "MuiTypography-body2-15",
                "button": "MuiTypography-button-18",
                "caption": "MuiTypography-caption-17",
                "colorError": "MuiTypography-colorError-30",
                "colorInherit": "MuiTypography-colorInherit-26",
                "colorPrimary": "MuiTypography-colorPrimary-27",
                "colorSecondary": "MuiTypography-colorSecondary-28",
                "colorTextSecondary": "MuiTypography-colorTextSecondary-29",
                "display1": "MuiTypography-display1-11",
                "display2": "MuiTypography-display2-10",
                "display3": "MuiTypography-display3-9",
                "display4": "MuiTypography-display4-8",
                "gutterBottom": "MuiTypography-gutterBottom-24",
                "headline": "MuiTypography-headline-12",
                "noWrap": "MuiTypography-noWrap-23",
                "paragraph": "MuiTypography-paragraph-25",
                "root": "MuiTypography-root-7",
                "subheading": "MuiTypography-subheading-14",
                "title": "MuiTypography-title-13",
              }
            }
            color="default"
            gutterBottom={true}
            headlineMapping={
              Object {
                "body1": "p",
                "body2": "aside",
                "display1": "h1",
                "display2": "h1",
                "display3": "h1",
                "display4": "h1",
                "headline": "h1",
                "subheading": "h3",
                "title": "h2",
              }
            }
            noWrap={false}
            paragraph={false}
            style={
              Object {
                "color": undefined,
              }
            }
            variant="body1"
          >
            <p
              className="MuiTypography-root-7 MuiTypography-body1-16 MuiTypography-gutterBottom-24 TextSection-sectionText-3"
              style={
                Object {
                  "color": undefined,
                }
              }
            />
          </Typography>
        </WithStyles(Typography)>
      </div>
    </div>
  </TextSection>
</WithStyles(TextSection)>
`;

exports[`TextSection should render a title when passed in 1`] = `
<WithStyles(TextSection)
  sectionTitle="Test Title"
>
  <TextSection
    classes={
      Object {
        "heroBtn": "TextSection-heroBtn-4",
        "innerContainer": "TextSection-innerContainer-2",
        "invertedBtn": "TextSection-invertedBtn-5",
        "section": "TextSection-section-1",
        "sectionText": "TextSection-sectionText-3",
        "white": "TextSection-white-6",
      }
    }
    sectionTitle="Test Title"
  >
    <div
      className="TextSection-section-1"
      style={
        Object {
          "backgroundColor": undefined,
          "borderBottom": "2px solid undefined",
          "borderTop": "2px solid undefined",
        }
      }
    >
      <div
        className="TextSection-innerContainer-2"
      >
        <WithStyles(Typography)
          gutterBottom={true}
          style={
            Object {
              "color": undefined,
            }
          }
          variant="display2"
        >
          <Typography
            align="inherit"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter-20",
                "alignJustify": "MuiTypography-alignJustify-22",
                "alignLeft": "MuiTypography-alignLeft-19",
                "alignRight": "MuiTypography-alignRight-21",
                "body1": "MuiTypography-body1-16",
                "body2": "MuiTypography-body2-15",
                "button": "MuiTypography-button-18",
                "caption": "MuiTypography-caption-17",
                "colorError": "MuiTypography-colorError-30",
                "colorInherit": "MuiTypography-colorInherit-26",
                "colorPrimary": "MuiTypography-colorPrimary-27",
                "colorSecondary": "MuiTypography-colorSecondary-28",
                "colorTextSecondary": "MuiTypography-colorTextSecondary-29",
                "display1": "MuiTypography-display1-11",
                "display2": "MuiTypography-display2-10",
                "display3": "MuiTypography-display3-9",
                "display4": "MuiTypography-display4-8",
                "gutterBottom": "MuiTypography-gutterBottom-24",
                "headline": "MuiTypography-headline-12",
                "noWrap": "MuiTypography-noWrap-23",
                "paragraph": "MuiTypography-paragraph-25",
                "root": "MuiTypography-root-7",
                "subheading": "MuiTypography-subheading-14",
                "title": "MuiTypography-title-13",
              }
            }
            color="default"
            gutterBottom={true}
            headlineMapping={
              Object {
                "body1": "p",
                "body2": "aside",
                "display1": "h1",
                "display2": "h1",
                "display3": "h1",
                "display4": "h1",
                "headline": "h1",
                "subheading": "h3",
                "title": "h2",
              }
            }
            noWrap={false}
            paragraph={false}
            style={
              Object {
                "color": undefined,
              }
            }
            variant="display2"
          >
            <h1
              className="MuiTypography-root-7 MuiTypography-display2-10 MuiTypography-gutterBottom-24"
              style={
                Object {
                  "color": undefined,
                }
              }
            >
              Test Title
            </h1>
          </Typography>
        </WithStyles(Typography)>
        <WithStyles(Typography)
          className="TextSection-sectionText-3"
          gutterBottom={true}
          style={
            Object {
              "color": undefined,
            }
          }
          variant="body1"
        >
          <Typography
            align="inherit"
            className="TextSection-sectionText-3"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter-20",
                "alignJustify": "MuiTypography-alignJustify-22",
                "alignLeft": "MuiTypography-alignLeft-19",
                "alignRight": "MuiTypography-alignRight-21",
                "body1": "MuiTypography-body1-16",
                "body2": "MuiTypography-body2-15",
                "button": "MuiTypography-button-18",
                "caption": "MuiTypography-caption-17",
                "colorError": "MuiTypography-colorError-30",
                "colorInherit": "MuiTypography-colorInherit-26",
                "colorPrimary": "MuiTypography-colorPrimary-27",
                "colorSecondary": "MuiTypography-colorSecondary-28",
                "colorTextSecondary": "MuiTypography-colorTextSecondary-29",
                "display1": "MuiTypography-display1-11",
                "display2": "MuiTypography-display2-10",
                "display3": "MuiTypography-display3-9",
                "display4": "MuiTypography-display4-8",
                "gutterBottom": "MuiTypography-gutterBottom-24",
                "headline": "MuiTypography-headline-12",
                "noWrap": "MuiTypography-noWrap-23",
                "paragraph": "MuiTypography-paragraph-25",
                "root": "MuiTypography-root-7",
                "subheading": "MuiTypography-subheading-14",
                "title": "MuiTypography-title-13",
              }
            }
            color="default"
            gutterBottom={true}
            headlineMapping={
              Object {
                "body1": "p",
                "body2": "aside",
                "display1": "h1",
                "display2": "h1",
                "display3": "h1",
                "display4": "h1",
                "headline": "h1",
                "subheading": "h3",
                "title": "h2",
              }
            }
            noWrap={false}
            paragraph={false}
            style={
              Object {
                "color": undefined,
              }
            }
            variant="body1"
          >
            <p
              className="MuiTypography-root-7 MuiTypography-body1-16 MuiTypography-gutterBottom-24 TextSection-sectionText-3"
              style={
                Object {
                  "color": undefined,
                }
              }
            />
          </Typography>
        </WithStyles(Typography)>
      </div>
    </div>
  </TextSection>
</WithStyles(TextSection)>
`;

exports[`TextSection should render a title when passed in 2`] = `
<WithStyles(TextSection)
  textColor="#000"
>
  <TextSection
    classes={
      Object {
        "heroBtn": "TextSection-heroBtn-4",
        "innerContainer": "TextSection-innerContainer-2",
        "invertedBtn": "TextSection-invertedBtn-5",
        "section": "TextSection-section-1",
        "sectionText": "TextSection-sectionText-3",
        "white": "TextSection-white-6",
      }
    }
    textColor="#000"
  >
    <div
      className="TextSection-section-1"
      style={
        Object {
          "backgroundColor": undefined,
          "borderBottom": "2px solid undefined",
          "borderTop": "2px solid undefined",
        }
      }
    >
      <div
        className="TextSection-innerContainer-2"
      >
        <WithStyles(Typography)
          gutterBottom={true}
          style={
            Object {
              "color": "#000",
            }
          }
          variant="display2"
        >
          <Typography
            align="inherit"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter-20",
                "alignJustify": "MuiTypography-alignJustify-22",
                "alignLeft": "MuiTypography-alignLeft-19",
                "alignRight": "MuiTypography-alignRight-21",
                "body1": "MuiTypography-body1-16",
                "body2": "MuiTypography-body2-15",
                "button": "MuiTypography-button-18",
                "caption": "MuiTypography-caption-17",
                "colorError": "MuiTypography-colorError-30",
                "colorInherit": "MuiTypography-colorInherit-26",
                "colorPrimary": "MuiTypography-colorPrimary-27",
                "colorSecondary": "MuiTypography-colorSecondary-28",
                "colorTextSecondary": "MuiTypography-colorTextSecondary-29",
                "display1": "MuiTypography-display1-11",
                "display2": "MuiTypography-display2-10",
                "display3": "MuiTypography-display3-9",
                "display4": "MuiTypography-display4-8",
                "gutterBottom": "MuiTypography-gutterBottom-24",
                "headline": "MuiTypography-headline-12",
                "noWrap": "MuiTypography-noWrap-23",
                "paragraph": "MuiTypography-paragraph-25",
                "root": "MuiTypography-root-7",
                "subheading": "MuiTypography-subheading-14",
                "title": "MuiTypography-title-13",
              }
            }
            color="default"
            gutterBottom={true}
            headlineMapping={
              Object {
                "body1": "p",
                "body2": "aside",
                "display1": "h1",
                "display2": "h1",
                "display3": "h1",
                "display4": "h1",
                "headline": "h1",
                "subheading": "h3",
                "title": "h2",
              }
            }
            noWrap={false}
            paragraph={false}
            style={
              Object {
                "color": "#000",
              }
            }
            variant="display2"
          >
            <h1
              className="MuiTypography-root-7 MuiTypography-display2-10 MuiTypography-gutterBottom-24"
              style={
                Object {
                  "color": "#000",
                }
              }
            />
          </Typography>
        </WithStyles(Typography)>
        <WithStyles(Typography)
          className="TextSection-sectionText-3"
          gutterBottom={true}
          style={
            Object {
              "color": "#000",
            }
          }
          variant="body1"
        >
          <Typography
            align="inherit"
            className="TextSection-sectionText-3"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter-20",
                "alignJustify": "MuiTypography-alignJustify-22",
                "alignLeft": "MuiTypography-alignLeft-19",
                "alignRight": "MuiTypography-alignRight-21",
                "body1": "MuiTypography-body1-16",
                "body2": "MuiTypography-body2-15",
                "button": "MuiTypography-button-18",
                "caption": "MuiTypography-caption-17",
                "colorError": "MuiTypography-colorError-30",
                "colorInherit": "MuiTypography-colorInherit-26",
                "colorPrimary": "MuiTypography-colorPrimary-27",
                "colorSecondary": "MuiTypography-colorSecondary-28",
                "colorTextSecondary": "MuiTypography-colorTextSecondary-29",
                "display1": "MuiTypography-display1-11",
                "display2": "MuiTypography-display2-10",
                "display3": "MuiTypography-display3-9",
                "display4": "MuiTypography-display4-8",
                "gutterBottom": "MuiTypography-gutterBottom-24",
                "headline": "MuiTypography-headline-12",
                "noWrap": "MuiTypography-noWrap-23",
                "paragraph": "MuiTypography-paragraph-25",
                "root": "MuiTypography-root-7",
                "subheading": "MuiTypography-subheading-14",
                "title": "MuiTypography-title-13",
              }
            }
            color="default"
            gutterBottom={true}
            headlineMapping={
              Object {
                "body1": "p",
                "body2": "aside",
                "display1": "h1",
                "display2": "h1",
                "display3": "h1",
                "display4": "h1",
                "headline": "h1",
                "subheading": "h3",
                "title": "h2",
              }
            }
            noWrap={false}
            paragraph={false}
            style={
              Object {
                "color": "#000",
              }
            }
            variant="body1"
          >
            <p
              className="MuiTypography-root-7 MuiTypography-body1-16 MuiTypography-gutterBottom-24 TextSection-sectionText-3"
              style={
                Object {
                  "color": "#000",
                }
              }
            />
          </Typography>
        </WithStyles(Typography)>
      </div>
    </div>
  </TextSection>
</WithStyles(TextSection)>
`;

exports[`TextSection should render text when passed in 1`] = `
<WithStyles(TextSection)
  text="Test text"
>
  <TextSection
    classes={
      Object {
        "heroBtn": "TextSection-heroBtn-4",
        "innerContainer": "TextSection-innerContainer-2",
        "invertedBtn": "TextSection-invertedBtn-5",
        "section": "TextSection-section-1",
        "sectionText": "TextSection-sectionText-3",
        "white": "TextSection-white-6",
      }
    }
    text="Test text"
  >
    <div
      className="TextSection-section-1"
      style={
        Object {
          "backgroundColor": undefined,
          "borderBottom": "2px solid undefined",
          "borderTop": "2px solid undefined",
        }
      }
    >
      <div
        className="TextSection-innerContainer-2"
      >
        <WithStyles(Typography)
          gutterBottom={true}
          style={
            Object {
              "color": undefined,
            }
          }
          variant="display2"
        >
          <Typography
            align="inherit"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter-20",
                "alignJustify": "MuiTypography-alignJustify-22",
                "alignLeft": "MuiTypography-alignLeft-19",
                "alignRight": "MuiTypography-alignRight-21",
                "body1": "MuiTypography-body1-16",
                "body2": "MuiTypography-body2-15",
                "button": "MuiTypography-button-18",
                "caption": "MuiTypography-caption-17",
                "colorError": "MuiTypography-colorError-30",
                "colorInherit": "MuiTypography-colorInherit-26",
                "colorPrimary": "MuiTypography-colorPrimary-27",
                "colorSecondary": "MuiTypography-colorSecondary-28",
                "colorTextSecondary": "MuiTypography-colorTextSecondary-29",
                "display1": "MuiTypography-display1-11",
                "display2": "MuiTypography-display2-10",
                "display3": "MuiTypography-display3-9",
                "display4": "MuiTypography-display4-8",
                "gutterBottom": "MuiTypography-gutterBottom-24",
                "headline": "MuiTypography-headline-12",
                "noWrap": "MuiTypography-noWrap-23",
                "paragraph": "MuiTypography-paragraph-25",
                "root": "MuiTypography-root-7",
                "subheading": "MuiTypography-subheading-14",
                "title": "MuiTypography-title-13",
              }
            }
            color="default"
            gutterBottom={true}
            headlineMapping={
              Object {
                "body1": "p",
                "body2": "aside",
                "display1": "h1",
                "display2": "h1",
                "display3": "h1",
                "display4": "h1",
                "headline": "h1",
                "subheading": "h3",
                "title": "h2",
              }
            }
            noWrap={false}
            paragraph={false}
            style={
              Object {
                "color": undefined,
              }
            }
            variant="display2"
          >
            <h1
              className="MuiTypography-root-7 MuiTypography-display2-10 MuiTypography-gutterBottom-24"
              style={
                Object {
                  "color": undefined,
                }
              }
            />
          </Typography>
        </WithStyles(Typography)>
        <WithStyles(Typography)
          className="TextSection-sectionText-3"
          gutterBottom={true}
          style={
            Object {
              "color": undefined,
            }
          }
          variant="body1"
        >
          <Typography
            align="inherit"
            className="TextSection-sectionText-3"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter-20",
                "alignJustify": "MuiTypography-alignJustify-22",
                "alignLeft": "MuiTypography-alignLeft-19",
                "alignRight": "MuiTypography-alignRight-21",
                "body1": "MuiTypography-body1-16",
                "body2": "MuiTypography-body2-15",
                "button": "MuiTypography-button-18",
                "caption": "MuiTypography-caption-17",
                "colorError": "MuiTypography-colorError-30",
                "colorInherit": "MuiTypography-colorInherit-26",
                "colorPrimary": "MuiTypography-colorPrimary-27",
                "colorSecondary": "MuiTypography-colorSecondary-28",
                "colorTextSecondary": "MuiTypography-colorTextSecondary-29",
                "display1": "MuiTypography-display1-11",
                "display2": "MuiTypography-display2-10",
                "display3": "MuiTypography-display3-9",
                "display4": "MuiTypography-display4-8",
                "gutterBottom": "MuiTypography-gutterBottom-24",
                "headline": "MuiTypography-headline-12",
                "noWrap": "MuiTypography-noWrap-23",
                "paragraph": "MuiTypography-paragraph-25",
                "root": "MuiTypography-root-7",
                "subheading": "MuiTypography-subheading-14",
                "title": "MuiTypography-title-13",
              }
            }
            color="default"
            gutterBottom={true}
            headlineMapping={
              Object {
                "body1": "p",
                "body2": "aside",
                "display1": "h1",
                "display2": "h1",
                "display3": "h1",
                "display4": "h1",
                "headline": "h1",
                "subheading": "h3",
                "title": "h2",
              }
            }
            noWrap={false}
            paragraph={false}
            style={
              Object {
                "color": undefined,
              }
            }
            variant="body1"
          >
            <p
              className="MuiTypography-root-7 MuiTypography-body1-16 MuiTypography-gutterBottom-24 TextSection-sectionText-3"
              style={
                Object {
                  "color": undefined,
                }
              }
            >
              Test text
            </p>
          </Typography>
        </WithStyles(Typography)>
      </div>
    </div>
  </TextSection>
</WithStyles(TextSection)>
`;
